### ISSUE TITLE
cmd/go: apparent livelock in TestScript/install_cmd_gobin via `setPGOProfilePath`

### ISSUE THREAD

**Issue by bcmills** [golang/go#60428]
```
#!watchflakes
post <- pkg == "cmd/go" && log ~ `install_cmd_gobin` && log ~ `^\s*cmd/go/internal/load\.setPGOProfilePath`
```

**Comment by prattmic** [comment]
It's not trivially reproducible on a gomote. I'll leave it going for a while and see if it makes progress. FWIW, this test takes ~150s to pass in race mode, pretty close to the timeout.

**Comment by bcmills** [comment]
I would expect the test to take quite a bit longer now that `cmd/compile` is built with a PGO profile.

However, most of that time _should_ be spent in rebuilding the imported packages, not in `load.PackagesAndErrors`. 🤔

**Comment by bcmills** [comment]
I suspect that something is going wrong with the `split` recursion:
https://cs.opensource.google/go/go/+/master:src/cmd/go/internal/load/pkg.go;l=2978-2982;drc=227e7126857df9f73863fc60090976b98fa9dd4e

Perhaps we're somehow ending up with O(N²) splits instead of O(N), or something like that?

**Comment by prattmic** [comment]
The fix for #60455 reduces the (non-race) run time of this test from ~7s to ~0.2s, so I am assume it is the same root cause.
